cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureUtils.java

```diff
@@ -291,6 +292,25 @@ public List<CloudVmInstanceStatus> deallocateInstances(AuthenticatedContext ac,
         return statusesAfterDeallocate;
     }
 
+    @Retryable(backoff = @Backoff(delay = 1000, multiplier = 2, maxDelay = 10000), maxAttempts = 5)
+    public List<CloudVmInstanceStatus> deallocateInstances(AuthenticatedContext ac, List<CloudInstance> vms, Long timeboundInMs) {
+        LOGGER.info("Deallocate VMs: {}", vms.stream().map(CloudInstance::getInstanceId).collect(Collectors.toList()));
+        List<CloudVmInstanceStatus> actualVmStatuses = azureVirtualMachineService.getVmsAndVmStatusesFromAzureWithoutRetry(ac, vms).getStatuses();
+        List<CloudVmInstanceStatus> statusesAfterDeallocate = new ArrayList<>();
+        List<CloudInstance> vmsNotStopped = new ArrayList<>();
+        actualVmStatuses.stream().forEach(instance -> {
+            if (InstanceStatus.STOPPED.equals(instance.getStatus())) {
```

Comment:
Filter out additional VM instances which may already be in a final state. (See the AwsInstanceConnector as an example).
As an example: If one of the instances is already in a DELETED state - there's no point attempting to STOP it.

Reply:
Done